parser_library/src/parsing/parser_impl.cpp

```diff
@@ -543,24 +543,24 @@ void parser_impl::transform_imm_reg_operands(semantics::collector& col, id_index
     instruction_name = *instruction;
     std::vector<std::pair<size_t, size_t>> replaced;
     auto opernds = &col.current_operands().value;
-    auto mnem_tmp = context::instruction::mnemonic_codes.find(*instruction);
-    if (mnem_tmp != context::instruction::mnemonic_codes.end())
+    if (auto mnem_tmp = context::instruction::mnemonic_codes.find(*instruction);
+        mnem_tmp != context::instruction::mnemonic_codes.end())
     {
         instruction_name = context::instruction::mnemonic_codes.at(*instruction).instruction->instr_name;
         replaced = context::instruction::mnemonic_codes.at(*instruction).replaced;
     }
```

Comment:
This could be simplified a bit to something like
```suggestion
    const machine_instruction* instr;
    const std::vector<std::pair<size_t, size_t>>* replaced = nullptr;
    if (auto mnem_tmp = context::instruction::mnemonic_codes.find(*instruction);
        mnem_tmp != context::instruction::mnemonic_codes.end())
    {
        instr = mnem_tmp->instruction;
        replaced = &mnem_tmp->replaced;
    }
    else
    {
        instr = &context::instruction::machine_instructions.at(*instruction);
    }
```
but it needs to be properly incorporated into the rest of the code